app/models/batch_connect/app.rb

```diff
@@ -220,7 +220,13 @@ def submit_file(root:, paths: %w(submit.yml.erb submit.yml))
       # Parse an ERB and Yaml file
       def read_yaml_erb(path:, binding: nil)
         contents = path.read
-        contents = ERB.new(contents, nil, "-").result(binding) if path.extname == ".erb"
+
+        if path.extname == ".erb"
+          erb = ERB.new(contents, nil, "-")
+          erb.filename = path.to_s
+          contents = erb.result(binding)
+        end
```

Comment:
I could probably add another method:

```ruby
def render_erb_file(path:, contents:, binding:)
  erb = ERB.new(contents, nil, "-")
  erb.filename = path.to_s
  erb.result(binding)
end
```

which could be useful/used in several places. By passing in path and contents as arguments (instead of having this new method read the contents from the path) this becomes a pure function.

Reply:
FWIW there is a one-liner that would allow this:

```ruby
ERB.new(contents, nil, "-").tap {|erb| erb.filename = path.to_s }. result(binding)
```

But that actually reduces readability over the multiline version.

Reply:
Agreed that the one liner is less readable in this situation.